pyro/infer/mcmc/nuts.py

```diff
@@ -162,30 +160,23 @@ def __init__(self,
         # Here, as suggested in [1], we set dE_max = 1000.
         self._max_sliced_energy = 1000
 
-    def _is_turning(self, r_left, r_right, r_sum):
+    def _is_turning(self, r_left_unscaled, r_right_unscaled, r_sum):
```

Comment:
I'm curious, does this mean the turning condition is defined in the original coordinates or in the adapted coordinates? I would have thought it would be defined in the adapted coordinates, as it is for reparameterized models.

Reply:
Originally, the turning condition is defined by `r_left . (z_right - z_left)`, which is approximated by `r_left . M^-1 . sum(r)` (because for each leapfrog step, `z' - z = M^-1 r'`). So if we unscale those `r`, it is `r_left_unscaled . sum(r_unscaled)`. I think we need to keep the definition of the turning condition.